### PR TITLE
 Render single media as albums.

### PR DESCRIPTION
**Pull request by charlesmchen-signal**
PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/ViewControllers/ConversationView/Cells/OWSMessageBubbleView.m [hunk 1]
```diff
@@ -795,47 +774,22 @@ - (UIView *)loadViewForMediaAlbum
...
     };
-    for (UIView *itemView in albumView.itemViews) {
-        OWSBubbleShapeView *strokeView = [[OWSBubbleShapeView alloc] initDraw];
-        strokeView.strokeColor = [UIColor colorWithWhite:0.5f alpha:0.4f];
-        strokeView.strokeThickness = 1.f;
-        [itemView addSubview:strokeView];
-        [self.bubbleView addPartnerView:strokeView];
-        [self.viewConstraints addObjectsFromArray:[strokeView ows_autoPinToSuperviewEdges]];
+
+    // Only apply "inner shadow" for single media, not albums.
```
Comment (michaelkirk-signal) [inline Signal/src/ViewControllers/ConversationView/Cells/OWSMessageBubbleView.m]: I thought the spec had shadows in albums too, was that an intentional design change?

File: Signal/src/ViewControllers/ConversationView/Cells/OWSMessageBubbleView.m [hunk 2]
```diff
@@ -1421,10 +1350,33 @@ - (void)handleMediaTapGesture
...
+        case OWSMessageCellType_MediaAlbum: {
             OWSAssertDebug(self.bodyMediaView);
             OWSAssertDebug(self.viewItem.mediaAlbumItems.count > 0);
 
+            if (self.viewItem.mediaAlbumHasFailedAttachment) {
+                [self.delegate didTapFailedIncomingAttachment:self.viewItem];
+                return;
+            }
+
+            // TODO: We might be able to get rid of this.
```
Comment (michaelkirk-signal) [inline Signal/src/ViewControllers/ConversationView/Cells/OWSMessageBubbleView.m]: Just for my own understanding re: this TODO, do you mean to get rid of it and replace it with the "did tap on specific album cell" behavior?

File: Signal/src/ViewControllers/ConversationView/Cells/OWSMessageBubbleView.m (hunk 1 above)
Comment (charlesmchen-signal) [reply Signal/src/ViewControllers/ConversationView/Cells/OWSMessageBubbleView.m]: I couldn't find out a good way to do it, so Myles approved taking them out.

File: Signal/src/ViewControllers/ConversationView/Cells/OWSMessageBubbleView.m (hunk 2 above)
Comment (charlesmchen-signal) [reply Signal/src/ViewControllers/ConversationView/Cells/OWSMessageBubbleView.m]: > Just for my own understanding re: this TODO, do you mean to get rid of it and replace it with the "did tap on specific album cell" behavior?

That's half of it. The other half of it is whether or not we need to special-case tapping on "lone" media v. "album" media.

**Pull request merged**